src/sphinxcontrib/bibtex/style/template.py

```diff
@@ -162,6 +162,75 @@ def reference(children, data: Dict[str, Any]):
     return SphinxReferenceText(info, *parts)
 
 
+# based on sphinx's make_refnode
+# to create citation_reference instead of reference
+def make_citation_reference_node(
+        builder: "Builder", fromdocname: str, todocname: str,
+        targetid: str,
+        child: Union[Node, List[Node]], title: str = None
+        ) -> docutils.nodes.citation_reference:
+    """Shortcut to create a citation reference node."""
+    # latex builder needs docname and refname
+    node = docutils.nodes.citation_reference('', '', internal=True)
+    if builder.name in {'latex', 'rinoh'}:
+        # builders using docname/refname
+        node['docname'] = todocname
+        node['refname'] = targetid
```

Comment:
@brechtm You could check here if rinoh builder needs special casing in the attributes it needs for the citation_reference to resolve correctly. I've assumed it works like the latex builder.